nntrainer/src/pooling2d_layer.cpp

```diff
@@ -59,9 +63,52 @@ Tensor Pooling2DLayer::forwarding(Tensor in, Tensor output, int &status) {
   return forwarding(in, status);
 }
 
-Tensor Pooling2DLayer::backwarding(Tensor in, int iteration) {
-  // NYI
-  return in;
+Tensor Pooling2DLayer::backwarding(Tensor derivative, int iteration) {
+  unsigned int batch = input_dim.batch();
+  unsigned int channel = input_dim.channel();
+  unsigned int height = input_dim.height();
+  unsigned int width = input_dim.width();
+  unsigned int p_height = pooling_size[0];
+  unsigned int p_width = pooling_size[1];
+  unsigned int p_size = p_height * p_width;
+
+  unsigned int I, J;
+  Tensor result = Tensor(input_dim);
+  float *out = result.getData();
+  switch (pooling_type) {
+  case PoolingType::max: {
+    for (unsigned int i = 0; i < derivative.getDim().getDataLen(); ++i) {
+      out[max_idx[i]] += derivative.getData()[i];
+    }
+  } break;
+  case PoolingType::average: {
+    for (unsigned int b = 0; b < batch; ++b) {
+      for (unsigned int i = 0; i < channel; ++i) {
+        I = 0;
+        for (unsigned int j = 0; j <= height - p_height; j += stride[0]) {
+          J = 0;
+          for (unsigned int k = 0; k <= width - p_width; k += stride[1]) {
```

Comment:
How about re-naming `I -> J` and `J -> K` as `I` and `J` are for height and width? At first I thought `I` is related to `i` and took me some time to verify this.

Reply:
fixed. Thanks